controllers/profile_controller.go

```diff
@@ -79,35 +94,56 @@ func (r *ProfileSubscriptionReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	pDef, err := git.GetProfileDefinition(pSub.Spec.ProfileURL, pSub.Spec.Branch, logger)
 	if err != nil {
-		r.patchStatusFailing(ctx, &pSub, logger, "error when fetching profile definition")
+		r.patchStatus(ctx, &pSub, logger, readyFalse, "FetchProfileFailed", "error when fetching profile definition")
 		return ctrl.Result{}, err
 	}
 
 	instance := profile.New(pDef, pSub, r.Client, logger)
-	err = instance.CreateArtifacts(ctx)
+	artifactStatus, err := instance.ArtifactStatus(ctx)
 	if err != nil {
-		r.patchStatusFailing(ctx, &pSub, logger, "error when creating profile artifacts")
 		return ctrl.Result{}, err
 	}
 
-	r.patchStatusRunning(ctx, &pSub, logger)
-	// TODO requeuing
-	return ctrl.Result{}, nil
-}
+	if !artifactStatus.ResourcesExist {
+		err = instance.CreateArtifacts(ctx)
+		if err != nil {
+			r.patchStatus(ctx, &pSub, logger, readyFalse, "CreateFailed", "error when creating profile artifacts")
+			return ctrl.Result{}, err
+		}
+		artifactStatus, err = instance.ArtifactStatus(ctx)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
 
-func (r *ProfileSubscriptionReconciler) patchStatusFailing(ctx context.Context, pSub *v1alpha1.ProfileSubscription, logger logr.Logger, message string) {
-	pSub.Status.State = "failing"
-	pSub.Status.Message = message
-	r.patchStatus(ctx, pSub, logger)
+	if len(artifactStatus.NotReadyConditions) == 0 {
+		r.patchStatus(ctx, &pSub, logger, readyTrue, "ArtifactsReady", "all artifact resouces ready")
+	} else {
+		var messages []string
+		status := readyUnknown
+		for _, condition := range artifactStatus.NotReadyConditions {
+			logger.Info(fmt.Sprintf("%s=%s, message:%s", condition.Type, string(condition.Status), condition.Message))
+			messages = append(messages, condition.Message)
+			if string(condition.Status) == readyFalse {
+				status = readyFalse
+			}
+		}
+		r.patchStatus(ctx, &pSub, logger, status, "ArtifactNotReady", strings.Join(messages, ","))
+	}
+	return ctrl.Result{}, nil
 }
 
-func (r *ProfileSubscriptionReconciler) patchStatusRunning(ctx context.Context, pSub *v1alpha1.ProfileSubscription, logger logr.Logger) {
-	pSub.Status.State = "running"
-	pSub.Status.Message = ""
-	r.patchStatus(ctx, pSub, logger)
-}
+func (r *ProfileSubscriptionReconciler) patchStatus(ctx context.Context, pSub *v1alpha1.ProfileSubscription, logger logr.Logger, readyStatus, reason, message string) {
```

Comment:
I'm wondering if this shouldn't return the possible errors it generates. Most of the places this is called from also bubble up errors anyways. WDYT?

Reply:
yeah its an interesting one, here if the patch fails we are in a weird state, the previous "actions" have succeeded, but we were not able to notify the resource it worked. Should this cause a failure? I think probably but its not 100% clear cut.

I will update the code to handle the error outside of the

Reply:
done